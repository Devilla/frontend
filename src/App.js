import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
