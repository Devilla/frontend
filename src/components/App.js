import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
