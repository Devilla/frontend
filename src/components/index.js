export { default as WebsiteHeader } from "./WebsiteHeader/WebsiteHeader";
export { default as WebsiteFooter } from "./WebsiteFooter/WebsiteFooter";
export { default as WebsiteHome } from "./WebsiteHome/WebsiteHome";
export { default as WebsiteHowItWorks } from "./WebsiteHowItWorks/WebsiteHowItWorks";
export { default as WebsiteIntegrations } from "./WebsiteIntegrations/WebsiteIntegrations";
export { default as WebsitePricing } from "./WebsitePricing/WebsitePricing";
export { default as WebsiteAbout } from "./WebsiteAbout/WebsiteAbout";
export { default as WebsiteSignIn } from "./WebsiteSignIn/WebsiteSignIn";
export { default as WebsiteSignUp } from "./WebsiteSignUp/WebsiteSignUp";
export { default as WebsiteContact } from "./WebsiteContact/WebsiteContact";
export { default as WebsiteTerms } from "./WebsiteTerms/WebsiteTerms";
export { default as WebsitePolicy } from "./WebsitePolicy/WebsitePolicy";
export { default as WebsiteDemoPage}  from "./WebsiteDemoPage/WebsiteDemoPage";
export { default as WebsiteCustStory}  from "./WebsiteCustStory/WebsiteCustStory";
export { default as WebsiteFeature}  from "./WebsiteFeature/WebsiteFeature";
export { default as Affiliate}  from "./Affiliate/Affiliate";
export { default as AffiliateRegister}  from "./Affiliate/AffiliateRegister";

export { default as Header } from "./Header/Header";
export { default as ForgetPassword } from "./ForgetPassword";
export { default as PricePage } from "./PricePage";
export { default as Common } from "./Common/function";
export { default as ResetPassword } from "./ResetPassword";
export { default as Sidebar } from "./Sidebar/Sidebar";
export { default as StatsCard } from "./StatsCard/StatsCard";
export { default as Dashboard } from './Dashboard/Dashboard';
export { default as Notification } from './Notification';
export { default as Analytics } from './Analytics/analytics';
export { default as AnalyticsProfile } from './Analytics/AnalyticsProfile';
export { default as New } from './New/new';
export { default as LoginFlow } from './LoginFlow/LoginFlow';
export { default as Spinner } from './Spinner/Spinner';
export { default as Error } from './Error/Error';
export { default as ComingSoon } from './ComingSoon/ComingSoon';
export { default as Button } from './Button/Button';
export { default as Profile } from './Profile/Profile';
export { default as BillingDetails } from './Billing/BillingDetails';
export { default as UpgradeCard } from './UpgradeCard/UpgradeCard';
export { default as UpgradePlan } from './UpgradePlan/UpgradePlan';
export { default as Integration } from './Integration/Integration';
export { default as Template } from './Template';
export { default as VerificationPage } from './VerificationPage/VerificationPage';
