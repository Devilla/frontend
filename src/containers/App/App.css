@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
