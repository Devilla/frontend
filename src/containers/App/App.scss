@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
