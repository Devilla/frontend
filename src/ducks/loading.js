import { Map } from 'immutable';

const action = name => `zikher/modals/${name}`;
const LOAD = action('LOAD');
const LOADED = action('LOADED');

export const load = () => ({ type: LOAD });
export const loaded = () => ({ type: LOADED });

const initialState = Map({
  state: false,
});

const loading = (state = initialState, action) => {
  switch (action.type) {
    case LOAD:
      return state.set('state', true);

    case LOADED:
      return state.set('state', false);

    default:
      return state;
  }
};

export default loading;
