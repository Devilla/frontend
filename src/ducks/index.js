import { combineReducers } from 'redux-immutable';
import { routerReducer } from 'react-router-redux'

import profile from './profile';
import auth from './auth';
import plan from './plan';
import payment from './payment';
import notification from './notification';
import campaign from './campaign';
import rules from './rules';
import loading from './loading';
import configuration from './configuration';
import elastic from './elastic';
import pageurl from './pageurl';
import webhooks from './webhooks';
import subcampaign from './subcampaign';

const reducer = combineReducers({
  profile,
  auth,
  plan,
  payment,
  notification,
  campaign,
  rules,
  loading,
  configuration,
  elastic,
  pageurl,
  webhooks,
  subcampaign,
  router: routerReducer
});

export default reducer;
