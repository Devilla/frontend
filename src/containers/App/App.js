import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
