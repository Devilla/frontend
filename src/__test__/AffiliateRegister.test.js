import React from 'react';
import '../setupTests';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import {createStore} from 'redux'
import { load, loaded } from 'ducks/loading';
import { AffiliateRegister } from 'components';
import { store } from 'App';

// Snapshot for Home React Component
describe('>>> AffiliateRegister --- Snapshot',()=>{
    it('+++capturing Snapshot of AffiliateRegister', () => {
        const renderedValue =  renderer.create(<AffiliateRegister store={store} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('>>>AffiliateRegister --- Shallow Render REACT COMPONENTS',()=>{
    let wrapper

    beforeEach(()=>{
        wrapper = shallow(<AffiliateRegister store={store}/>)

    })

    it('+++ render the AffiliateRegister component', () => {
       expect(wrapper.length).toEqual(1)
}


it('+++ should respond to change event and change the state of the affiliate register Component', () => {
    wrapper.find('input').simulate('change', {target: {name: 'name', value: 'JOHNDOE'}});
    expect(wrapper.state('name')).toEqual('JOHNDOE');
   })

   // it('+++ should respond to change event and change the state of the affiliate register Component', () => {
   //  wrapper.find('#email').simulate('change', {target: {name: 'email', value: 'JOHNDOE@gmail.com' }});
   //  expect(wrapper.state('email')).toEqual('JOHNDOE@gmail.com');
   // })


// it('+++ contains button with id="affiliatesubmit"', () => {
//         expect(wrapper.find('button#affiliatesubmit').type()).toEqual('button')
// });
//
//
//    it('should stimulate the click event' ,() => {
//     wrapper.find('#affiliatesubmit').simulate('click');
//    })
//
//    it("+++ fake dummy values for the affilate fields >>> should fail" ,() => {
//        wrapper.find('#name').simulate('change',{ target: {name: 'name', value: 12121}});
//        expect(wrapper.state('name')).toBe(12121);
//    })
//    it("+++ fake dummy values for the affilate fields >>> should fail" ,() => {
//     wrapper.find('#email').simulate('change',{ target: {name: 'email', value: '12121$gmail.com'}});
//     expect(wrapper.state('email')).toBe('12121$gmail.com');
// })
//
//
//
//
});
