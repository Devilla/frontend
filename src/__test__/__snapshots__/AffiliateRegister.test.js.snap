// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>T E R M S --- Snapshot +++capturing Snapshot of terms 1`] = `
<div
  className="affiliateregister-container"
>
  <div
    className="main-container"
  >
    <section>
      <div
        className="container"
      >
        <div
          className="row justify-content-around"
        >
          <h2
            className="align-center"
          >
            Become an Influence Affiliate
          </h2>
          <div
            className="col-md-6"
          >
            <form
              className="row"
              data-name="Affiliate Form"
              method="POST"
              onSubmit={[Function]}
            >
              <div
                className="mt-3 col-md-12"
              >
                <input
                  className="ml-0 validate-required"
                  id="name"
                  name="name"
                  onChange={[Function]}
                  placeholder="First / Last Name"
                  type="text"
                />
                <span
                  className="help-block"
                >
                  <p
                    className="website-error"
                  >
                    
                  </p>
                </span>
              </div>
              <div
                className="mt-3 col-md-12"
              >
                <input
                  className="field w-input"
                  name="email"
                  onChange={[Function]}
                  placeholder="Johndoe@example.com"
                  type="email"
                />
                <span
                  className="help-block"
                >
                  <p
                    className="website-error"
                  >
                    
                  </p>
                </span>
              </div>
              <div
                className="mt-3 col-md-12"
              >
                <input
                  className="button submit-button w-button btn btn--primary ml-0"
                  disabled={true}
                  style={
                    Object {
                      "cursor": "pointer",
                    }
                  }
                  type="submit"
                  value="BECOME A AFFILIATE"
                />
              </div>
              <div />
            </form>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
